cNVMe/Tests.cpp

```diff
@@ -268,9 +269,82 @@ namespace cnvme
 				pDriverCommand->Command.DWord0Breakdown.OPC = 0xFE; // invalid.. hopefully
 
 				driver.sendCommand(buffer, BUF_SIZE);
+
+				auto status = pDriverCommand->CompletionQueueEntry.SC;
+				delete[] buffer;
 				
-				ASSERT_IF(pDriverCommand->CompletionQueueEntry.SC != constants::status::codes::generic::INVALID_COMMAND_OPCODE, "Controller did not fail invalid opcode correctly");
+				ASSERT_IF(status != constants::status::codes::generic::INVALID_COMMAND_OPCODE, "Controller did not fail invalid opcode correctly");
+
+				return true;
+			}
+
+			bool testNVMeQueueDeletionFailures()
+			{
+				cnvme::driver::Driver driver;
+
+				UINT_32 BUF_SIZE = sizeof(cnvme::driver::DRIVER_COMMAND);
+				BYTE* buffer = new BYTE[BUF_SIZE];
+				memset(buffer, 0, BUF_SIZE);
+
+				auto pDriverCommand = (cnvme::driver::PDRIVER_COMMAND)buffer;
+				pDriverCommand->QueueId = ADMIN_QUEUE_ID;
+
+				UINT_32 timeout = 5; // arbitrary
+				pDriverCommand->Timeout = timeout;
+				pDriverCommand->TransferDataDirection = cnvme::driver::NO_DATA;
+				pDriverCommand->Command.DWord0Breakdown.OPC = constants::opcodes::admin::DELETE_IO_COMPLETION_QUEUE;
+				pDriverCommand->Command.DW10_DeleteIoQueue.QID = ADMIN_QUEUE_ID;
+
+				driver.sendCommand(buffer, BUF_SIZE);
+
+				auto status = pDriverCommand->CompletionQueueEntry.SC;
+				ASSERT_IF(status != constants::status::codes::specific::INVALID_QUEUE_IDENTIFIER, "Controller did not fail deleting the admin queue correctly");
```

Comment:
Maybe reword this to clarify the error: "Expected controller to fail IO queue deletion with admin queue ID, but did not receive INVALID_QUEUE_IDENTIFIER status."